distribution/test/itests/test-itests-common/src/test/java/ddf/common/test/ServiceManager.java

```diff
@@ -233,10 +233,32 @@ public void stopFeature(boolean wait, String... featureNames) throws Exception {
     }
 
     // TODO - we should really make this a bundle and inject this.
-    private FeaturesService getFeaturesService() {
-        ServiceReference<FeaturesService> featuresServiceRef = bundleCtx.getServiceReference(
-                FeaturesService.class);
-        return bundleCtx.getService(featuresServiceRef);
+    private FeaturesService getFeaturesService() throws InterruptedException {
+        FeaturesService featuresService = null;
+        boolean ready = false;
+        long timeoutLimit = System.currentTimeMillis() + REQUIRED_BUNDLES_TIMEOUT;
+        while (!ready) {
+            ServiceReference<FeaturesService> featuresServiceRef = bundleCtx.getServiceReference(
+                    FeaturesService.class);
+            try {
+                if (featuresServiceRef != null) {
+                    featuresService = bundleCtx.getService(featuresServiceRef);
+                    ready = true;
```

Comment:
You could do your null check on featuresService here so callers of this method know it will always return the features service.